controllers/cart.js

```diff
@@ -0,0 +1,59 @@
+const { products, carts, orders } = require('../storage');
+const crypto = require('crypto');
+
+function createCart(id) {
+  if (!carts.find((item) => item.userId === id)) {
+    const cart = {
+      id: crypto.randomUUID(),
+      userId: id,
+      products: [],
+    };
+    carts.push(cart);
+  }
+}
+
+function addProductToCart(id, product) {
+  const thisProduct = products.find((item) => item.id === +product);
+  const thisCart = carts.find((item) => item.userId === id);
+  thisCart.products.push(thisProduct);
+}
+
+const updateCart = (req, res) => {
+  const headerUserId = req.headers['x-user-id'];
+  const { productId } = req.params;
+  createCart(headerUserId);
+  addProductToCart(headerUserId, productId);
+
+  res.status(200).send(carts);
+};
+
+const deleteProductFromCart = (req, res) => {
+  const headerUserId = req.headers['x-user-id'];
+  const { productId } = req.params;
+  const thisCart = carts.find((item) => item.userId === headerUserId);
+  thisCart.products.filter((item) => item.id !== +productId);
```

Comment:
the `filter` method creates a new array, it doesn't modify initial one. So you need to assign the result of `filter` to the initial array
`thisCart.products = thisCart.products.filter((item) => item.id !== +productId);`